app/src/main/java/com/ece493/group5/adjustableaudio/MainActivity.java

```diff
@@ -1,182 +1,200 @@
-            }
-        });
-
-        alertDialogBuilder.setNegativeButton(R.string.negative_button_dialog, new DialogInterface.OnClickListener()
-        {
-            @Override
-            public void onClick(DialogInterface dialogInterface, int i)
-            {
-                dialogInterface.cancel();
-            }
-        });
-        alertDialogBuilder.show();
-    }
-}
+package com.ece493.group5.adjustableaudio;
+
+
+import android.Manifest;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.media.browse.MediaBrowser;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.MenuItem;
+import android.widget.Toast;
+
+import com.ece493.group5.adjustableaudio.listeners.EqualizerModelListener;
+import com.ece493.group5.adjustableaudio.models.EqualizerModel;
+import com.ece493.group5.adjustableaudio.services.MicrophoneService;
+import com.ece493.group5.adjustableaudio.services.MusicService;
+import com.google.android.material.bottomnavigation.BottomNavigationView;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Objects;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+import androidx.navigation.NavController;
+import androidx.navigation.Navigation;
+import androidx.navigation.ui.AppBarConfiguration;
+import androidx.navigation.ui.NavigationUI;
+
+public class MainActivity extends AppCompatActivity implements EqualizerModelListener
+{
+    private static final String TAG = MainActivity.class.getSimpleName();
+
+    private static final int REQUEST_CODE_PERMISSIONS = 1;
+
+    private EqualizerModel equalizerModel;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        BottomNavigationView navView = findViewById(R.id.nav_view);
+        // Passing each menu ID as a set of Ids because each
+        // menu should be considered as top level destinations.
+        AppBarConfiguration appBarConfiguration = new AppBarConfiguration.Builder(
+                R.id.navigation_hearing_test, R.id.navigation_microphone, R.id.navigation_media_player, R.id.navigation_settings)
+                .build();
+        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
+        NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);
+        NavigationUI.setupWithNavController(navView, navController);
+
+        equalizerModel = new EqualizerModel(getApplicationContext());
+
+        checkAndRequestPermissions();
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
```

Comment:
Inconsistent curly brace. As well, the Android documentation says that "item" should never be null. Should there be a null check here?

Reply:
I added a null check and it highlighted my text with a warning. I think maybe it never being null means that android ensures that it isn't?